### PR TITLE
Refresh eventing user docs

### PR DESCRIPTION
**Pull request by nlopezgi**
Part of https://github.com/knative/eventing/issues/3541

## Proposed Changes 

- Migrate content from https://docs.google.com/document/d/1pL-p3Urm7RCWt_ZDbhMM52sMpMUhO3L-UFeaEXO4FiI/edit# to knative eventing docs
- This PR currently leaves some links as TBD as I want to first get some agreement that this is the right place to put this content.
-

fyi @grantr @abrennan89

### CODE REVIEW

File: docs/eventing/README.md [hunk 1]
```diff
@@ -3,6 +3,65 @@ Knative Eventing is a system that is designed to address a common need for cloud
...
+Microservices, and functions, usually get their input from something else. That
+“something else” is the trigger for a microservice to take care of business. The
+piece of information that is sent across is called an event. The microservices
+in an event-driven architecture are constantly reacting to events and producing
+events themselves. As an example, think about your version control system. When
+you make changes to a project and want to merge it into the main branch, you
+need to create a Pull Request (PR). When you create that PR, it could trigger a
+bunch of other actions like code linting, quality tests, and update tickets.
+All those actions are started by events and the actions consume (act on) that
+data.
```
Comment (abrennan89) [inline docs/eventing/README.md:28]: I'd also remove this in favor or just providing a link to something like https://cloudevents.io/ in the very first paragraph - to me this explanation doesn't make it any clearer what events are, and the title is sort of misleading.

File: docs/eventing/README.md [hunk 2]
```diff
@@ -3,6 +3,65 @@ Knative Eventing is a system that is designed to address a common need for cloud
 native development and provides composable primitives to enable late-binding
 event sources and event consumers.
```
Comment (abrennan89) [inline docs/eventing/README.md:4]:
```suggestion
event sources and event consumers.

Knative Eventing is modeled on the concept of event-driven architecture. For more information about events, see the [CloudEvents](https://cloudevents.io/) documentation.
```

File: docs/eventing/README.md [hunk 3]
```diff
@@ -3,6 +3,65 @@ Knative Eventing is a system that is designed to address a common need for cloud
...
+bunch of other actions like code linting, quality tests, and update tickets.
+All those actions are started by events and the actions consume (act on) that
+data.
+
+#### Traditional Pub/Sub Model
+To get event data to consumers, one model you can use is the traditional
+“Pub/Sub” model. This model was made popular with the rise of
+service-oriented architecture, where events are placed on channels
+(or “topics”) and all subscribers receive are notified. Modern tools that fit
+in this category are Amazon SNS, Azure Service Bus, and Google Cloud Pub/Sub.
```
Comment (abrennan89) [inline docs/eventing/README.md:35]:
```suggestion
## Supported event delivery models

- Pub/Sub using Channels and Subscriptions
- Event-driven functions, using event sources
- Event mesh, using Brokers and Triggers
```

File: docs/eventing/README.md [hunk 4]
```diff
@@ -3,6 +3,65 @@ Knative Eventing is a system that is designed to address a common need for cloud
...
+hosted Function-as-a-Service platforms like AWS Lambda, Azure Functions, and
+Google Cloud Function.
+
+#### Modern Event Mesh
+As the serverless industry evolved and technology matured, the industry started
+to build architectures where the producers and consumers were even more
+separated. At the same time, the delivery of the events is governed by
+high-level routing and event data is captured in a standard format (like Cloud
+Events). These architectures are facilitated by so called brokers like AWS
+EventBridge and Azure Event Grid.
```
Comment (abrennan89) [inline docs/eventing/README.md:50]: Delete.

File: docs/eventing/README.md [hunk 5]
```diff
@@ -3,6 +3,65 @@ Knative Eventing is a system that is designed to address a common need for cloud
...
+
+
+* When these things happen do these things and then some (N:M:Z event delivery): So, there’s a lot of things we can do with the constructs introduced in the use cases above, but the one downside with them is that for more complex use cases, where there are a multitude of Sources sending events, Functions consuming/transforming and so forth, it can get a bit unwieldy to keep track of which Channel is handling which events. Furthermore, sometimes you might only want to consume only some events, but with the constructs introduced so far, you would have to receive all of them, and throw out the ones you’re not interested in. To make these kinds of interactions easier and allow the user to only focus on declaring which events they are interested in and where to send them, there is an easier way to reason about them. This is where _Broker_ and _Trigger_ are meant to provide a straightforward user experience. (Read more - coming soon)
```
Comment (abrennan89) [inline docs/eventing/README.md:63]: I like the idea of having a better list of use cases on the Eventing landing page, but I think this again is much too wordy and doesn't really explain the different use case in a way that is valuable.

**Comment by nlopezgi** [comment]
> Hey @nlopezgi
> 
> Thanks for taking a stab at this. Overall I applaud the desire to improve the Eventing docs, but honestly I'm not sure that any of the information added from the Google doc in its current state is helpful.
> 
> It seems very wordy / casual, and I don't think it necessarily captures any real Knative Eventing specific info or explains the use cases particularly well.
> 
> I think maybe we should follow this up with async discussion on Slack or if you'd like to attend the docs WG we can hash it out?
> 
> Thanks again for your input, looking forward to discussing with you further!

Thanks Ashleigh for the quick review! I'm closing this PR as this is likely not the right direction to solve https://github.com/knative/eventing/issues/3541. Let's sync over slack to see if there is a way that I can help with the issue as part of the docathon effort that I'm involved in today and tomorrow. I'll try to attend next week's docs WG meeting, but I'm not sure at this moment if I can commit to continuing work on documentation beyond the current docathon effort.
